src/main/java/org/folio/services/ftp/FtpUploadService.java

```diff
@@ -110,47 +88,62 @@ private Handler<AsyncResult<Object>> asyncResultHandler(Promise<String> promise,
     };
   }
 
-  public Future<String> upload(Context ctx, String filename, String content) {
+  private Handler<AsyncResult<String>> asyncResult(Promise<String> promise) {
+    return result -> {
+      if (result.succeeded()) {
+        logger.debug("Success upload to FTP");
+        promise.complete(result.result());
+      } else {
+        String message = Optional.ofNullable(result.cause())
+          .map(Throwable::getMessage)
+          .orElse("Failed upload to FTP");
+        logger.error(message);
+        promise.fail(result.cause());
+      }
+    };
+  }
+
+  public Future<String> upload(Context ctx, String username, String password, String filename, String content) {
     Promise<String> promise = Promise.promise();
-    ctx.owner().executeBlocking(blockingFeature -> {
-      try (InputStream is = new ByteArrayInputStream(content.getBytes())) {
-        ftp.setFileType(FTP.BINARY_FILE_TYPE);
-        ftp.enterLocalPassiveMode();
-        changeWorkingDirectory();
-        if (ftp.storeFile(filename, is)) {
-          logger.debug("Batch voucher uploaded on FTP");
-          blockingFeature.complete(ftp.getReplyString().trim());
-        } else {
-          blockingFeature.fail(new FtpException(ftp.getReplyCode(), ftp.getReplyString().trim()));
-        }
-      } catch (Exception e) {
-        logger.error("Error uploading", e);
-        blockingFeature.fail(new CompletionException(e));
+    ctx.owner().executeBlocking(blockingFeature -> login(username, password).compose(ftpClient -> {
+    try (InputStream is = new ByteArrayInputStream(content.getBytes())) {
+      ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+      ftpClient.enterLocalPassiveMode();
+      changeWorkingDirectory(ftpClient);
+      if (ftpClient.storeFile(filename, is)) {
+        logger.info("Batch voucher uploaded on FTP");
```

Comment:
try to add as much identifiers to logs as possible, please revisit all logs in this PR to add more identifiers,
also formatting here should be fixed, try(InputStream is ...) - should be formatted with 2 spaces

Reply:
fixed